wasm/src/record/record_plaintext.rs

```diff
@@ -43,6 +43,17 @@ impl RecordPlaintext {
     pub fn gates(&self) -> u64 {
         ***self.0.gates()
     }
+
+    #[wasm_bindgen(js_name = serialNumber)]
+    pub fn serial_number(&self, private_key: &str, program_id: &str, record_name: &str) -> String {
```

Comment:
The private key optimally shouldn't be passed in as a string as this implies that on the javascript side, the private key is stored in JS memory as a string which is high-risk if we're assuming systems might be compromised. I'd suggest passing this in as the wasm-type. 

```suggestion
    pub fn serial_number(&self, private_key: PrivateKey, program_id: &str, record_name: &str) -> String {
```

Reply:
Ultimately this should likely have a path for private key in ciphertext form wherein the private key ciphertext is decrypted within the function, but we can address this as further iteration.